tgui/packages/tgui/interfaces/AppearanceChanger.js

```diff
@@ -0,0 +1,310 @@
+import { useBackend } from "../backend";
+import { LabeledList, Button } from "../components";
+import { Window } from "../layouts";
+
+export const AppearanceChanger = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    change_race,
+    species,
+    specimen,
+    change_gender,
+    gender,
+    has_gender,
+    change_eye_color,
+    change_skin_tone,
+    change_skin_color,
+    change_head_accessory_color,
+    change_hair_color,
+    change_secondary_hair_color,
+    change_facial_hair_color,
+    change_secondary_facial_hair_color,
+    change_head_marking_color,
+    change_body_marking_color,
+    change_tail_marking_color,
+    change_head_accessory,
+    head_accessory_styles,
+    head_accessory_style,
+    change_hair,
+    hair_styles,
+    hair_style,
+    change_facial_hair,
+    facial_hair_styles,
+    facial_hair_style,
+    change_head_markings,
+    head_marking_styles,
+    head_marking_style,
+    change_body_markings,
+    body_marking_styles,
+    body_marking_style,
+    change_tail_markings,
+    tail_marking_styles,
+    tail_marking_style,
+    change_body_accessory,
+    body_accessory_styles,
+    body_accessory_style,
+    change_alt_head,
+    alt_head_styles,
+    alt_head_style,
+  } = data;
+
+  let has_colours = false;
+  // Yes this if statement is awful, but I would rather do it here than inside the template
+  if (change_eye_color || change_skin_tone || change_skin_color || change_head_accessory_color || change_hair_color || change_secondary_hair_color || change_facial_hair_color || change_secondary_facial_hair_color || change_head_marking_color || change_body_marking_color || change_tail_marking_color) {
+    has_colours = true;
+  }
+
+  return (
+    <Window>
+      <Window.Content scrollable>
+        <LabeledList>
+          {!!change_race && (
+            <LabeledList.Item label="Species">
+              {species.map(s => (
+                <Button
+                  key={s.specimen}
+                  content={s.specimen}
+                  selected={s.specimen === specimen}
+                  onClick={
+                    () => act('race', { race: s.specimen })
+                  }
+                />
+              ))}
+            </LabeledList.Item>
+          )}
+          {!!change_gender && (
+            <LabeledList.Item label="Gender">
```

Comment:
Nitpick. Can you turn these LabledLists into their own method (class... thing not sure how it's called in js)?
Makes it a bit more readable.

Reply:
I cant think of a pheasible way to do this, but youre welcome to improve it later